### DISCUSSION TITLE
Passing binaries release through GitHub Desktop

### DISCUSSION THREAD

**Discussion by adslbarxatov** [community/community#23680] [New to GitHub]
<p>Hi!</p>
<p>How can we push release (binaries / executables) through Windows version of GitHub Desktop? Cannot find <em>understandable</em> instructions.</p>
<p>Thanks!</p>

**Comment by francisfuzz** [comment]
<aside class="quote no-group" data-username="adslbarxatov" data-post="1" data-topic="137131">
<div class="title">
<div class="quote-controls"></div>
 adslbarxatov:</div>
<blockquote>
<p>How can we push release (binaries / executables) through Windows version of GitHub Desktop? Cannot find <em>understandable</em> instructions.</p>
</blockquote>
</aside>
<p>:wave: <a class="mention" href="/adslbarxatov">@adslbarxatov</a>: GitHub Desktop tracks all changes to all files as you edit them, be it plaintext files or binaries (executables). You can follow the same instructions for <a href="https://docs.github.com/en/free-pro-team@latest/desktop/contributing-and-collaborating-using-github-desktop/committing-and-reviewing-changes-to-your-project">committing and reviewing changes to your project</a> and <a href="https://docs.github.com/en/free-pro-team@latest/desktop/contributing-and-collaborating-using-github-desktop/pushing-changes-to-github">pushing changes to GitHub</a>. If that doesn’t help, could you share an example of what you’re looking to accomplish?</p>

**Comment by adslbarxatov** [comment]
<p>Thanks for your answer!</p>
<p>Sorry, sorry, maybe, we have described our question oddly a bit. We need to  <em>draft new release</em>  through GitHub Desktop. Just as it is in GitHub on web. Is it possible?</p>
<p>We learned how to create tags. But that doesn’t seem like a solution we need</p>

**Comment by francisfuzz** [comment]
<aside class="quote no-group" data-username="adslbarxatov" data-post="4" data-topic="137131">
<div class="title">
<div class="quote-controls"></div>
 adslbarxatov:</div>
<blockquote>
<p>Sorry, sorry, maybe, we have described our question oddly a bit. We need to <em>draft new release</em> through GitHub Desktop. Just as it is in GitHub on web. Is it possible?</p>
<p>We learned how to create tags. But that doesn’t seem like a solution we need</p>
</blockquote>
</aside>
<p>:bulb: Thanks for clarifying! I have a better understanding of your question.</p>
<p>It’s not possible to draft or create a new release in GitHub Desktop in the <a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/managing-releases-in-a-repository#creating-a-release">same way that you can in the GitHub.com user interface</a> or <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-a-release">creating a release via the GitHub API</a>. GitHub Desktop is open source and has a <a href="https://github.com/desktop/desktop#how-can-i-contribute-to-github-desktop">contributing guide</a> that might be worth checking out if you’d like to propose this idea in an issue or in a pull request. Alternatively, you could set up an application or create a <a href="https://lab.github.com/githubtraining/github-actions:-hello-world">GitHub Actions workflow</a> to automate the creation of a new release given some event. I hope this helps!</p>

**Comment by adslbarxatov** [comment]
<p>Thanks for answer!</p>
<p>Sad, but OK.</p>
<p>We started to use Actions (thanks!). We’ve broken our minds, but created our first script. It works almost correct (<a href="https://github.com/adslbarxatov/DPModule/blob/794bc3629509c99a6e60d9713c3bd8124d64184b/.github/workflows/Release.yaml" rel="noopener nofollow ugc">https://github.com/adslbarxatov/DPModule/blob/794bc3629509c99a6e60d9713c3bd8124d64184b/.github/workflows/Release.yaml</a>).<br>
But we have no idea, how to pass environment variable from line 42 to line 52 (into <em>actions/github-script@v2 section</em>). (And we almost sure, that there is the same trouble in line 58). Execution results give no info, just <em>unexpected string</em>, without even the line number.</p>
<p><em>actions/upload-release-asset</em> failed to parse file name wildcards. So, we cannot use it. <em>actions/github-script@v2</em> described as universal solution for troubles like this. And we saw you as its contributor. May be, you can help us to finish this?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="adslbarxatov" data-post="6" data-topic="137131">
<div class="title">
<div class="quote-controls"></div>
 adslbarxatov:</div>
<blockquote>
<p>But we have no idea, how to pass environment variable from line 42 to line 52 (into <em>actions/github-script@v2 section</em> ).</p>
</blockquote>
</aside>
<p><code>$RID</code> isn’t a valid Javascript environment variable reference as far as I’m aware. You should be able to do the Actions style substitution in the script snippet without using the environment variable (<code>${{ steps.create_release.outputs.id }}</code>), or if that doesn’t work try the Node way of using <a href="https://nodejs.org/api/process.html#process_process_env"><code>process.env</code></a>.</p>

**Comment by adslbarxatov** [comment]
<p>Thanks!</p>
<p>Seems like <code>${{}}</code> syntax works.<br>
But the next problem occurs in the path parameter: where are <code>fs.readdir</code> and <code>fs.readfile</code> pointing when they get <code>'.'</code> path? It is not the repo root folder, because they ‘say’ that there are no such files (we used <code>./Release/${file}</code> pattern). But they are there. What is our mistake?</p>

**Comment by adslbarxatov** [comment]
<p>P. S. <code>/Release</code> pattern doesn’t point to <code>&lt;repo_root&gt;/Release</code> directory too. We’re stuck…</p>

**Comment by airtower-luna** [comment]
<p>You need to check out the repository using <a href="https://github.com/actions/checkout"><code>actions/checkout</code></a> before you can use files from it, otherwise your workflow has just an empty workspace. The simplest case, to get the latest commit on the branch the workflow is running on:</p>
<pre><code class="lang-yaml">      - uses: actions/checkout@v2
</code></pre>
<p>I assume you might want to fetch a release tag instead, in that case you can use the <code>ref</code> parameter for <code>actions/checkout</code>:</p>
<pre><code class="lang-yaml">      - uses: actions/checkout@v2
        with:
          ref: ${{ env.TAG }}
</code></pre>

**Comment by adslbarxatov** [comment, marked as answer]
<p>Thank you!</p>
<p>It looks like we won.</p>
<p>I was grown on ANSI C and C#, where <code>"</code> is a string / array and <code>'</code> is a character. Everything is simple and clear. And <code>\x60</code> (reversed apostroph or something) as <em>another one string delimiter that can dereference variables</em>… It’s weird a bit…</p>
<p>But it works. So, thank you for help!</p>
